sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -412,31 +470,42 @@ static RunQueryRequest makeRequest(Query query, @Nullable String namespace) {
 
     @Override
     public PCollection<Entity> expand(PBegin input) {
-      V1Options v1Options = V1Options.from(getProjectId(), getQuery(),
-          getNamespace());
+      V1Options v1Options = V1Options.from(getProjectId(), getNamespace());
 
       /*
        * This composite transform involves the following steps:
-       *   1. Create a singleton of the user provided {@code query} and apply a {@link ParDo} that
-       *   splits the query into {@code numQuerySplits} and assign each split query a unique
-       *   {@code Integer} as the key. The resulting output is of the type
-       *   {@code PCollection<KV<Integer, Query>>}.
+       *   1. Create a singleton of the user provided {@code query} or if {@code gqlQuery} is
+       *   provided apply a {@link ParDo} that translates the {@code gqlQuery} into a {@code query}.
+       *
+       *   2. A {@link ParDo} splits the resulting query into {@code numQuerySplits} and
+       *   assign each split query a unique {@code Integer} as the key. The resulting output is
+       *   of the type {@code PCollection<KV<Integer, Query>>}.
        *
        *   If the value of {@code numQuerySplits} is less than or equal to 0, then the number of
        *   splits will be computed dynamically based on the size of the data for the {@code query}.
        *
-       *   2. The resulting {@code PCollection} is sharded using a {@link GroupByKey} operation. The
+       *   3. The resulting {@code PCollection} is sharded using a {@link GroupByKey} operation. The
        *   queries are extracted from they {@code KV<Integer, Iterable<Query>>} and flattened to
        *   output a {@code PCollection<Query>}.
        *
-       *   3. In the third step, a {@code ParDo} reads entities for each query and outputs
+       *   4. In the third step, a {@code ParDo} reads entities for each query and outputs
        *   a {@code PCollection<Entity>}.
        */
-      PCollection<KV<Integer, Query>> queries = input
-          .apply(Create.of(getQuery()))
+
+      PCollection<Query> inputQuery;
+      if (getQuery() != null) {
+        inputQuery = input.apply(Create.of(getQuery()));
+      } else {
+        inputQuery = input
+            .apply(Create.of(getGqlQuery())
+                .withCoder(SerializableCoder.of(new TypeDescriptor<ValueProvider<String>>() {})))
+            .apply(ParDo.of(new GqlQueryTranslatorFn(v1Options)));
```

Comment:
Naming of fns is inconsistent: e.g. the split one is called SplitQueryFn (DoSomethingFn), and this one is called GqlQueryTranslatorFn (SomethingDoerFn). Please make them consistent. Arguably, DoSomethingFn is the better style.